packages/flutter_tools/lib/src/web/compile.dart

```diff
@@ -16,96 +16,112 @@ import '../globals.dart' as globals;
 import '../platform_plugins.dart';
 import '../plugins.dart';
 import '../project.dart';
+import '../reporting/reporting.dart';
+import '../version.dart';
 import 'migrations/scrub_generated_plugin_registrant.dart';
 
 export '../build_system/targets/web.dart' show kDart2jsDefaultOptimizationLevel;
 
-Future<void> buildWeb(
-  FlutterProject flutterProject,
-  String target,
-  BuildInfo buildInfo,
-  bool csp,
-  String serviceWorkerStrategy,
-  bool sourceMaps,
-  bool nativeNullAssertions,
-  bool isWasm, {
-  String dart2jsOptimization = kDart2jsDefaultOptimizationLevel,
-  String? baseHref,
-  bool dumpInfo = false,
-  bool noFrequencyBasedMinification = false,
-  String? outputDirectoryPath,
-}) async {
-  final bool hasWebPlugins = (await findPlugins(flutterProject))
-    .any((Plugin p) => p.platforms.containsKey(WebPlugin.kConfigKey));
-  final Directory outputDirectory = outputDirectoryPath == null
-      ? globals.fs.directory(getWebBuildDirectory(isWasm))
-      : globals.fs.directory(outputDirectoryPath);
-  outputDirectory.createSync(recursive: true);
+class WebBuilder {
+  WebBuilder({
+    required Logger logger,
+    required BuildSystem buildSystem,
+    required Usage usage,
+    required FlutterVersion flutterVersion,
+    required FileSystem fileSystem,
+  })  : _logger = logger,
+        _buildSystem = buildSystem,
+        _flutterUsage = usage,
+        _flutterVersion = flutterVersion,
+        _fileSystem = fileSystem;
 
-  // The migrators to apply to a Web project.
-  final List<ProjectMigrator> migrators = <ProjectMigrator>[
-    ScrubGeneratedPluginRegistrant(flutterProject.web, globals.logger),
-  ];
+  final Logger _logger;
+  final BuildSystem _buildSystem;
+  final Usage _flutterUsage;
+  final FlutterVersion _flutterVersion;
+  final FileSystem _fileSystem;
 
-  final ProjectMigration migration = ProjectMigration(migrators);
-  migration.run();
+  Future<void> buildWeb(
```

Comment:
Most of this method is whitespace changes, which you can optionally ignore in the GitHub UI for this change, if you want:
<img width="429" alt="Screenshot 2023-04-04 at 7 10 32 PM" src="https://user-images.githubusercontent.com/682784/229963177-6eda42a0-7c0c-4800-abd7-befc8f9acc30.png">